src/i965_gpe_utils.c

```diff
@@ -1406,6 +1406,27 @@ i965_object_surface_to_2d_gpe_resource(struct i965_gpe_resource *res,
 }
 
 void
+i965_object_surface_to_2d_gpe_resource_with_align(struct i965_gpe_resource *res,
+                                                  struct object_surface *obj_surface)
+{
+    unsigned int swizzle;
```

Comment:
keep i965_object_surface_to_2d_gpe_resource() and don't modify it. But update obj_surface->orig_width and  obj_surface->orig_width to new aligned value